pandas/core/dimensioned.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+
+from pandas.core.base import (PandasObject)
+from pandas.util._decorators import cache_readonly
+from pandas import compat
+from pandas.core.common import is_null_slice
+
+
+class Dimensional(PandasObject):
+    """
```

Comment:
you need to step thru code where we use an existing dtype and see how its done. its not easy but all is there. start with simple tests.